src/Abc.Zebus/Core/Bus.cs

```diff
@@ -888,10 +885,10 @@ private string DumpMessageOnDisk(MessageTypeId messageTypeId, Stream messageStre
 
                 var dumpFileName = $"{_deserializationFailureTimestampProvider.NextUtcTimestamp():yyyyMMdd-HH-mm-ss.fffffff}_{messageTypeId.FullName}";
                 var dumpFilePath = Path.Combine(DeserializationFailureDumpDirectoryPath, dumpFileName);
-                messageStream.Seek(0, SeekOrigin.Begin);
                 using (var fileStream = new FileStream(dumpFilePath, FileMode.Create))
                 {
-                    messageStream.CopyTo(fileStream);
+                    var messageBytes = messageContent.ToArray();
+                    fileStream.Write(messageBytes, 0, messageBytes.Length);
```

Comment:
Yes, I noticed that and I did not even consider using double-targeting because this code is non-critical.
It might be a good idea to add double-targeting to allow more important optimizations though.